### PR TITLE
fastly: init at v2.0.0

### PR DESCRIPTION
**Pull request by shyim**
###### Description of changes

Added the Fastly CLI. Currently, there are only a terraform provider and a Prometheus exporter

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/fastly/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    installShellCompletion fastly.{bash,zsh}
+
+    mkdir -p $out/bin
+    install fastly $out/bin/fastly
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "fastly is an open-source command line tool for interacting with the Fastly API.";
```
Comment (fabaff) [inline pkgs/misc/fastly/default.nix:56]:
```suggestion
    description = "Command line tool for interacting with the Fastly API";
```

see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

File: pkgs/misc/fastly/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchurl, writeShellScript, lib, installShellFiles, curl, jq, common-updater-scripts }:
+
+let
+  pname = "fastly";
+  version = "v2.0.0";
```
Comment (fabaff) [inline pkgs/misc/fastly/default.nix:5]:
```suggestion
  version = "2.0.0";
```

Comment (shyim) [reply pkgs/misc/fastly/default.nix]: fixed :)

File: pkgs/misc/fastly/default.nix (hunk 1 above)
Comment (shyim) [reply pkgs/misc/fastly/default.nix]: ups. fixed :)

**Review by ereslibre — APPROVED** [review]
Although this can be bumped to 3.1.0 already, +1. Can be bumped after this merges if desired.

**Comment by ereslibre** [comment]
Result of `nixpkgs-review pr 170077` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>fastly</li>
  </ul>
</details>

**Comment by ereslibre** [comment]
Result of `nixpkgs-review pr 170077` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>fastly</li>
  </ul>
</details>

**Review by ereslibre — CHANGES_REQUESTED** [review]
Sources are unfetchable, it has to be bumped.

**Pull request closed without merging**